chance.js

```diff
@@ -926,6 +927,22 @@
                 min: min,
                 max: max
             });
+        } if (options && (options.minAge || options.maxAge)) {
+            testRange(options.minAge > options.maxAge, "Chance: minAge cannot be greater than maxAge.");
+
+            var minAge = typeof options.minAge !== "undefined" ? options.minAge : 1;
+            var maxAge = typeof options.maxAge !== "undefined" ? options.maxAge : 100;
+
+            var minDate = new Date(now.getFullYear() - maxAge - 1, now.getMonth(), now.getDate());
+            var maxDate = new Date(now.getFullYear() - minAge, now.getMonth(), now.getDate());
```

Comment:
I suspect you may have edge cases here when `minAge` is 0 where you could get a birthday in the future

For example if it is January 1, 2022, then the valid range of birthdays for 0 would be January 1, 2021-January 1, 2022

But I suspect here you'd get something in the range January 1, 2022 - December 31, 2022

I would suggest writing some unit tests to test these gnarly edge cases

Reply:
hey, @victorquinn, your comment relates to a doubt I have about tests in Chance. As `chance.birthday()` only calculates boundaries and call `chance.date()`, my first thought was _spying_ on date and assessing the arguments it actually receives. I saw no _mocks/stubs/spies_ on the test suite, though, so I wonder: would you be ok with this approach?

About this particular case, the boundaries are correct, as below:
```js
// test setup
const now=new Date(2022,0,1)
const minAge=0
const maxAge=0

// code copied from implementation
const currentYear = now.getFullYear()
let minDate = new Date(currentYear - maxAge - 1, now.getMonth(), now.getDate())
let maxDate = new Date(currentYear - minAge, now.getMonth(), now.getDate());

maxDate.setDate(maxDate.getDate() +1);
maxDate.setMilliseconds(maxDate.getMilliseconds() -1);

// assessment of the date limits
console.log(minDate) // 2021-01-01T00:00:00.000Z
console.log(maxDate) // 2022-01-01T23:59:59.999Z
```

Reply:
I just pushed a version with tests without mocks that were able to find some corner cases. Maybe this should not be so important in this context.